Neos.Flow/Classes/Core/Booting/Scripts.php

```diff
@@ -736,9 +740,32 @@ public static function buildPhpCommand(array $settings)
             $command .= ' -c ' . escapeshellarg($useIniFile);
         }
 
+        self::ensureCurrentPhpVersionMatchesConfiguredVersion($command);
+
         return $command;
     }
 
+    /**
+     * @param $command
+     * @throws Exception
+     */
+    protected static function ensureCurrentPhpVersionMatchesConfiguredVersion($command)
+    {
+        $runningPHPVersion = phpversion() . '-' . PHP_SAPI;
+        exec($command . ' -r "echo phpversion() . \'-\' . PHP_SAPI;"', $output, $result);
```

Comment:
This would run an additional exec each time. Didn't you mean we could check against the `PHP_BINARY` to avoid that?

Reply:
Yes, I can do that too. We can't compare the versions in that case, but that's not necessary I guess.